R/serializer.R

```diff
@@ -662,18 +662,25 @@ createGraphicsDevicePromiseDomain <- function(which = dev.cur()) {
       force(onRejected)
       function(...) {
         old <- dev.cur()
-        dev.set(which)
-        on.exit(dev.set(old))
+        dev_set(which)
+        on.exit(dev_set(old))
 
         onRejected(...)
       }
     },
     wrapSync = function(expr) {
       old <- dev.cur()
-      dev.set(which)
-      on.exit(dev.set(old))
+      dev_set(which)
+      on.exit(dev_set(old))
 
       force(expr)
     }
   )
 }
+    
+dev_set <- function(i) {
+  # make sure to not open a device when calling `dev.set(1)`
+  if (i > 1) {
+    dev.set(i)
+  }
```

Comment:
Let's add a warning here (and maybe drop the stopifnot above?) just so we can be aware of this happening if it actually does (it shouldn't)